python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK.py

```diff
@@ -0,0 +1,406 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Created with the Rule Development Kit: https://github.com/awslabs/aws-config-rdk
+# Can be used stand-alone or with the Rule Compliance Engine: https://github.com/awslabs/aws-config-engine-for-compliance-as-code
+#
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  AMI_NOT_PUBLIC_CHECK
+
+Description:
+  Check whether the Amazon Machine Images are not publicly accessible. The rule is NON_COMPLIANT if one or more Amazon Machine Images are publicly accessible.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::::Account
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+    Given: No AMIs with "is-public" parameter set to True
+     Then: Return NOT_APPLICABLE
+  Scenario: 2
+    Given: One or more AMIs with is-public parameter set to True
+     Then: Return NON_COMPLIANT with Annotation containing AMIIDs
+'''
+
+import json
+import datetime
+import botocore
+import boto3
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+# Generates non-compliant response
+def generate_evaluation_list(images, event):
+    evaluations = []
+    # Looping through all available images
+    for image in images:
+        evaluation = build_evaluation(
+            image['ImageId'],
+            "NON_COMPLIANT",
+            event,
+            resource_type=DEFAULT_RESOURCE_TYPE,
+            annotation="Amazon Machine Image Id: {} is public".format(image['ImageId'])
+        )
+        evaluations.append(evaluation)
+    return evaluations
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
```

Comment:
Delete those explanation for readability

Reply:
Removed